### PR TITLE
Android Keystore Fixes

### PR DESCRIPTION
**Pull request by MarcoPolo**
A couple things:

1. Don't eat errors. This will return the actual errors to go which will get logged. This should help debug future errors. Before we were catching errors and rethrowing a generic error – lame.

1. Add a warning screen if user is android version <=22 to tell them they need a lock screen or they'll have to type their password everytime (see screenshot)

1. Fix a bug that happens on versions >22 that results in the secret store not working because there is a permanently invalid key. This can happen if you originally have a lock screen and then remove it.

@patrickxb @keybase/react-hackers 
screen shot 2017-10-12 at 2 35 01 pm

### CODE REVIEW

File: shared/constants/platform.native.js [hunk 1]
```diff
@@ -7,12 +7,18 @@ NativeModules.ObjcEngine || {
...
+  isDeviceSecure: 'fallback',
 }
 const isStoryBook = (NativeModules.Storybook && NativeModules.Storybook.isStorybook) || false
 const version = nativeBridge.version
 const appVersionName = nativeBridge.appVersionName
 const appVersionCode = nativeBridge.appVersionCode
 const isSimulator = nativeBridge.usingSimulator === '1'
+// Currently this is given to us as a boolean, but no real documentation on this, so just in case it changes in the future.
+// Android only field that tells us if there is a lock screen.
+const isDeviceSecureAndroid: boolean = typeof nativeBridge.isDeviceSecure === 'boolean'
```
Comment (chrisnojima) [inline shared/constants/platform.native.js]: We do enforce this in the java side so I don’t think we have to be worried about it that much. This is the sole consumer so I’m not worried about it being out of sync

Comment (MarcoPolo) [reply shared/constants/platform.native.js]: I was worried about the RN side since it's type is just Object.

**Comment by chrisnojima** [comment]
Yeah but this comes from us. Not rn directly right

File: shared/constants/platform.native.js (hunk 1 above)
Comment (MarcoPolo) [reply shared/constants/platform.native.js]: It works through RN. Something like this:
```
 @Override
  public Map<String, Object> getConstants() {
    final Map<String, Object> constants = new HashMap<>();
    constants.put(DURATION_SHORT_KEY, Toast.LENGTH_SHORT);
    constants.put(DURATION_LONG_KEY, Toast.LENGTH_LONG);
    return constants;
  }
```

I was a little paranoid that at some future point RN could convert the values to strings.

File: shared/react-native/android/app/src/main/java/io/keybase/ossifrage/modules/KeybaseEngine.java [hunk 2]
```diff
@@ -100,12 +102,21 @@ public String getName() {
     public Map<String, Object> getConstants() {
         String versionCode = String.valueOf(BuildConfig.VERSION_CODE);
         String versionName = BuildConfig.VERSION_NAME;
+        boolean isDeviceSecure = false;
```
Comment (chrisnojima) [inline shared/react-native/android/app/src/main/java/io/keybase/ossifrage/modules/KeybaseEngine.java]: But it’s a Boolean here so won’t java complain

**Pull request merged**